src/sensor/pingsensor.h

```diff
@@ -197,19 +197,19 @@ class PingSensor : public Sensor
      */
     void writeMessage(const ping_message& msg) const;
 
-    QString _ascii_text;
-    uint8_t _device_revision;
-    uint8_t _device_type;
-    uint8_t _dstId;
-    uint8_t _firmware_version_major;
-    uint8_t _firmware_version_minor;
-    uint8_t _firmware_version_patch;
-    int _lostMessages;
-    QString _nack_msg;
-    uint8_t _protocol_version_major;
-    uint8_t _protocol_version_minor;
-    uint8_t _protocol_version_patch;
-    uint8_t _srcId;
+    QString _ascii_text = QString();
```

Comment:
QStrings are not necessary here, the default constructors do that, besides this forces the unnecessary creation and move of an empty string.

Reply:
What about the zero for integers? Is this different?

I thought maybe it is a problem with the dynamic creation of the object.

> this forces the unnecessary creation and move of an empty string.

I think @tcanabrava has mentioned QString is smart enough that this doesn't happen.

Reply:
it's different: *if* QStrings where a class without default constructor, then nothing would happen as the compiler wouldn't know what to do and garbage would be inside the values. But QString is a class with a default constructor that creates an empty string. int, on the other hand, has no default value initializer = you can call int a{0}; to default initialize it to 0, but if you just do a `int i;` there's gargage in memory.

Reply:
perfect :100: